### PR TITLE
REF: Rewrite Introduction with more free text.

### PR DESCRIPTION
**Pull request by clukas1**
From latex intro

### CODE REVIEW

File: Writerside/topics/introduction.md [hunk 1]
```diff
@@ -1,97 +1,82 @@
...
+The primary research question addressed by this thesis is how the RAPTOR algorithm can be implemented and validated
+within the context of the Swiss public transit network. This question is particularly relevant given the complexity and
+scale of the network, which poses significant challenges for routing algorithms. A related research question focuses on
+the optimization of the RAPTOR algorithm in Java and C++ and comparing the performance of both programming environments
+for such a task.
+
+These research questions are framed within the context of the current state of public transit networks. The complexity
+of these networks, which involve multiple routes, stops, footpaths, and transfer points, necessitates the development of
+routing algorithms that account for factors such as transfer times, service frequencies, and real-time constraints. The
+research seeks to address these challenges while also considering scalability, as expanding transit systems further
```
Comment (munterfi) [inline Writerside/topics/introduction.md:41]: I don't get this sentence... expanding in what manner?

File: Writerside/topics/introduction.md [hunk 2]
```diff
@@ -1,97 +1,82 @@
...
+routing algorithm, enabling a modular architecture. The modularity allows for system components to be modified or
+replaced without affecting external dependencies, ensuring the system's adaptability and long-term maintainability.
+
+To facilitate interaction with the system, a REST API will be developed, allowing external applications to access the
+routing and scheduling services. This API will provide the foundation for seamless integration between the routing
+algorithm and external systems. As a proof of concept, a simple web application will be created to visualize search
+results and demonstrate the capabilities of the backend service. While the web application will serve as a proof of
+concept, the primary focus of this thesis remains on developing the backend service.
+
+This approach ensures the development of a scalable, high-performance public transit routing package based on the RAPTOR
```
Comment (munterfi) [inline Writerside/topics/introduction.md:71]: Maybe remove this paragraph? It is kind of a repetition of the above?

File: Writerside/topics/introduction.md [hunk 3]
```diff
@@ -1,97 +1,82 @@
...
+
+These research questions are framed within the context of the current state of public transit networks. The complexity
+of these networks, which involve multiple routes, stops, footpaths, and transfer points, necessitates the development of
+routing algorithms that account for factors such as transfer times, service frequencies, and real-time constraints. The
+research seeks to address these challenges while also considering scalability, as expanding transit systems further
+complicates the problem space.
+
+## Approach and Expected Outcomes
+
+This thesis adopts a comprehensive approach to address the identified challenges, beginning with the utilization of GTFS
```
Comment (munterfi) [inline Writerside/topics/introduction.md:46]: Remove the second sentence? Or merge with the first one? The detail of a GTFS schedule with trips, routes and stop times is not yet important in the context of the introduction.

File: Writerside/topics/introduction.md [hunk 4]
```diff
@@ -1,97 +1,82 @@
...
+concept, the primary focus of this thesis remains on developing the backend service.
+
+This approach ensures the development of a scalable, high-performance public transit routing package based on the RAPTOR
+algorithm, as well as a flexible service component capable of managing complex transit schedules and routes. The end
+result is a modular system, complemented by a user-friendly interface that showcases the potential of the underlying
+backend service.
+
+## Scope and Delimitation
+
+This thesis is carried out as part of a student project, with the primary objective of advancing knowledge and
```
Comment (munterfi) [inline Writerside/topics/introduction.md:78]: Maybe a bit shorter, and also omit the intention of the commercial product, I think this is clear given the part `no external company interests`:

```
This thesis is part of the Master of Advanced Studies in Software Engineering at the Eastern Switzerland University of
Applied Sciences (OST). It focuses on applying the knowledge gained from the program to further advance methodologies in
public transit routing, with no external company interests influencing the scope or outcomes of this research.
```

**Review by munterfi — CHANGES_REQUESTED** [review]
Thanks @clukas1 

Looks good, but i would keep it a bit shorter. Therefore I have made some comments.

File: Writerside/topics/introduction.md [hunk 5]
```diff
@@ -1,97 +1,82 @@
...
-   considering time-dependent constraints.
-
-3. **Public Transit Service**: This integration layer adds abstraction between the publicly visible interface and the
-   core components, such as the routing algorithm and schedule data source. This will allow for exchanging those
-   components without interfering with any outside dependencies on the service.
-
-4. **Web Application**: Around the core routing functionality, we develop a simple web application. The frontend will
-   allow users to explore schedules, query connections, and assess accessibility. The backend continuously updates the
-   static schedule.
+This thesis is driven by several key challenges in public transit routing. In recent years, many public transit agencies
```
[the remark this answers is not in the document]
Comment (clukas1) [reply Writerside/topics/introduction.md:10]: 0a3dc1b

File: Writerside/topics/introduction.md [hunk 6]
```diff
@@ -1,97 +1,82 @@
...
+have begun publishing their schedules using the General Transit Feed Specification (GTFS) format as open data [3].
+Maintaining accurate public transit schedules is a resource-intensive task. The sheer volume and complexity of
```
[the remark this answers is not in the document]
Comment (clukas1) [reply Writerside/topics/introduction.md:12]: 0a3dc1b

File: Writerside/topics/introduction.md [hunk 7]
```diff
@@ -1,97 +1,82 @@
...
+GTFS data present significant challenges for developing efficient data structures and processing methodologies.
+
+Public transit routing is inherently complex due to the time-dependent nature of transit networks. Transit schedules
+vary based on the time of day, day of the week, and other temporal factors, adding an additional layer of complexity to
```
[the remark this answers is not in the document]
Comment (clukas1) [reply Writerside/topics/introduction.md:16]: 0a3dc1b

File: Writerside/topics/introduction.md [hunk 8]
```diff
@@ -1,97 +1,82 @@
...
+the routing process. Developing efficient algorithms that account for this time-dependency remains a challenging task in
+transportation research. Traditional graph-based routing algorithms often face performance issues when dealing with
+large-scale transit networks, leading to very large graph structures and computational cost in preparing such graphs.
```
[the remark this answers is not in the document]
Comment (clukas1) [reply Writerside/topics/introduction.md:19]: What is the difference of preparation and preprocessing?

File: Writerside/topics/introduction.md [hunk 9]
```diff
@@ -1,97 +1,82 @@
...
+The RAPTOR (Round-based Public Transit Routing) algorithm, recognized for its speed and accuracy, offers a promising
```
[the remark this answers is not in the document]
Comment (clukas1) [reply Writerside/topics/introduction.md:20]: 0a3dc1b

File: Writerside/topics/introduction.md (hunk 1 above)
Comment (clukas1) [reply Writerside/topics/introduction.md:41]: I was thinking of multi-modal or more trip related filters, but I removed it for now.

0a3dc1b

File: Writerside/topics/introduction.md (hunk 2 above)
Comment (clukas1) [reply Writerside/topics/introduction.md:71]: Agreed, 0a3dc1b

File: Writerside/topics/introduction.md (hunk 3 above)
Comment (clukas1) [reply Writerside/topics/introduction.md:46]: 0a3dc1b

File: Writerside/topics/introduction.md (hunk 4 above)
Comment (clukas1) [reply Writerside/topics/introduction.md:78]: 0a3dc1b

**Pull request merged**